### PR TITLE
Legend assistant

### PR DESCRIPTION
**Pull request by jrmi**

### CODE REVIEW

File: src/modules/RA/DataLayer/components/DataLayerForm.js [hunk 1]
```diff
@@ -20,75 +20,80 @@ const DataLayerForm = ({ ...props }) => {
...
+        .filter(Boolean);
+      const fieldsInError = fields.filter((f, index) => f && fieldIndexes.includes(index));
+
+      let inError = false;
+      if (fieldsInError.length > 0) {
+        inError = true;
+      }
+      if (minisheetConfig && touched['minisheet_config.wizard.title']) {
+        inError = true;
       }
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/DataLayerForm.js:74]: This part of code is a kind of "duplicate" of the function before.
Maybe you should DRY this code with its own function

File: src/modules/RA/DataLayer/components/tabs/LegendTab/index.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        </Button>
+      </Placeholder>
+    );
+  }
+
+  return (
+    <>
+      <ul className={classes.legends}>
+        {legends && legends.map((legend, index) => (
+          <li className="legend">
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/tabs/LegendTab/index.js:96]: No `key` prop?

File: src/services/react-admin/patchLayerDataProvider.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      return itemRest;
+    });
+
+  return compatible;
+};
+
+
+const patchLayerDataProvider = nextDataProvider => async (...args) => {
+  const [type, resource, params, meta = {}] = args;
+  const { endpoint = resource } = meta;
```
Comment (dtrucs) [inline src/services/react-admin/patchLayerDataProvider.js:48]: Why don't you spread all theses variables directly in the arguments part?

**Review by dtrucs — CHANGES_REQUESTED** [review]

File: src/modules/RA/DataLayer/components/DataLayerForm.js (hunk 1 above)
Comment (jrmi) [reply src/modules/RA/DataLayer/components/DataLayerForm.js]: Code not modified anymore.

File: src/modules/RA/DataLayer/components/tabs/LegendTab/index.js (hunk 2 above)
Comment (jrmi) [reply src/modules/RA/DataLayer/components/tabs/LegendTab/index.js]: Right. Was at bad place.

File: src/services/react-admin/patchLayerDataProvider.js (hunk 3 above)
Comment (jrmi) [reply src/services/react-admin/patchLayerDataProvider.js]: Copy paste :)
Fixed.

File: src/services/react-admin/patchLayerDataProvider.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { GET_ONE  } from 'react-admin';
+import { RES_DATALAYER } from '../../modules/RA/ra-modules';
+
+// Patch legend from old standard to new one.
+const patchLegend = ({ stackedCircles, content, source, items = [], ...rest }) => {
+  const compatible = { ...rest };
+
+  compatible.shape = rest.shape;
```
Comment (dtrucs) [inline src/services/react-admin/patchLayerDataProvider.js:8]: useless?

Comment (jrmi) [reply src/services/react-admin/patchLayerDataProvider.js]: Fixed

**Review by dtrucs — APPROVED** [review]

**Review by CFarcy — APPROVED** [review]

**Pull request merged**